department-of-veterans-affairs/caseflow
# Pamatya/report type status bug

Resolves [Jira Issue Title](https://jira.devops.va.gov/browse/JIRA-12345)
No Ticket was created for this bug.

# Description
Please explain the changes you made here.

1. Added a yup.lazy validation to ignore the required validation if TimingSpecification is not defined. 

## Acceptance Criteria
- [ ] Validation for timing specification should work as expected
- [ ] When you select report-type: Status, it should still generate data in console.log.

## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-12345) or list them below
2.  Go to ReportPage as an admin
3. Select Report Type = status, See if Generate Report button is enabled. 
4. Click on Generate Report button, right click on the page and go to inspect to view the console. you should be able to see a console log with data {reportType: 'status'}
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/122557351/eb2da086-9901-4ae3-9e45-0b7bed8d738d)
5. Click on the Report_type and change report type to Event / action, Timing Specification should be visible. Validate all the validation for timing specification. 
a)  When Range is After/Before, Date picker should not be blank and error message should be -> Please enter a valid date, date should not be in future. 
b) when Rage is between, Date Picker should not be blank error message should be -> Please enter a valid date. 
From/To should not be in future, and To date cannot be greater than from date. 


# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---

## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
